src/tests.rs

```diff
@@ -988,3 +986,109 @@ fn add_where_item() {
     assert_quote_snapshot!(basic_type);
     assert_quote_snapshot!(type_with_args);
 }
+
+// =================
```

Comment:
Looks like you went the extra mile to consider multiple cases for the impl parser. I definitely appreciate it!